src/item.cpp

```diff
@@ -8724,11 +8725,17 @@ bool item::use_charges( const itype_id &what, int &qty, std::list<item> &used,
         } else if( e->count_by_charges() ) {
             if( e->typeId() == what ) {
 
+                // Unseal the pocket and parent pockets.
+                if( parent != nullptr ) {
+                    map_cursor mc( pos );
+                    avatar &player = get_avatar();
+                    item_location container( mc, parent );
+                    handle_contents_changed_helper handler( player, item_location( container, e ) );
+                    handler.handle();
```

Comment:
Calling `handle` may remove the content or container, but the content item `e` still need to be accessed later.

Reply:
Very true. Will need to reconsider how to approach this.

Reply:
The handler may remove the item (or its container) if it is spillable. What happens if you craft with for example a tin can of broth in your inventory?

Reply:
Did some tests with a bucket of clean water used to craft sweet water. 

Water bucket wielded - works as expected, and water is kept in the bucket with no prompt to spill it.

In backpack using 'w' to sheath the bucket, which certainly looks like a bug in itself. Bucket of water prompts for a container to spill it into, and then the sweet water prompts for a destination container. No debugmsg or problems encountered.

Reply:
I think it's because `container` is a map item location, so when the handler handles `e`, it thinks the container of `e` is on the map and so `e` does not need handling. Then it simply unseals `parent`, and since `container` is a map location, it decides that `parent` does not have any further parents and returns. `Character::consume_items` then calls `empty_buckets`, which is where the containers are actually handled. The net effect is that calling `handle()` here is only equivalent to calling `unseal` on the parent's pocket that contains `e`.

Reply:
So by that description all that needs to be done is to unseal the pocket. Which greatly simplifies things.